crates/llm-local/src/lib.rs

```diff
@@ -107,18 +106,13 @@ impl LlmEngine for LocalLlmEngine {
 }
 
 impl LocalLlmEngine {
-    pub async fn new(registry: PathBuf, use_gpu: bool) -> Self {
-        let mut engine = Self {
+    pub fn new(registry: PathBuf, use_gpu: bool) -> Self {
+        Self {
             registry,
             use_gpu,
             inferencing_models: Default::default(),
             embeddings_models: Default::default(),
-        };
-
-        let _ = engine.inferencing_model("llama2-chat".into()).await;
```

Comment:
I had to delete this because initialization is now synchronous  - we prefer to initialize factors as fast as possible. This was trying to bubble up missing model errors to initialization time. They now move to first usage.

Reply:
If we feel like retaining this validation is worthwhile we could make both `LlmFactor::new` and `TriggerFactors::new` async and fallible and check there.

Reply:
Unfortunately, the engine constructed in `LlmFactor::new` but rather in `Factor::configure_app` since the engines are configured by the runtime config. We would have to make `Factor::configure_app` async which I think we're less willing to do?

Reply:
I suppose we could also do whatever checks we like in `spin-runtime-config`... :thinking: